packages/flutter/test/material/tabs_test.dart

```diff
@@ -3950,6 +3950,141 @@ void main() {
     expect(iconTheme.color, equals(selectedTabColor));
   });
 
+  testWidgets('TabBar colors labels correctly', (WidgetTester tester) async {
+    int color = 0x00000000;
+    MaterialStateColor buildMSC() {
+      final Color selectedColor = Color(color++);
+      final Color unselectedColor = Color(color++);
+      return MaterialStateColor
+          .resolveWith((Set<MaterialState> states) {
+            if (states.contains(MaterialState.selected)) {
+              return selectedColor;
+            }
+            return unselectedColor;
+          });
+    }
+
+    final Color materialLabelColor = buildMSC();
+    final Color labelColor = Color(color++);
+    final Color unselectedLabelColor = Color(color++);
+
+    // this is to make sure labelStyles (in TabBar and in TabBarTheme) don't
+    // affect label's color. for details: https://github.com/flutter/flutter/pull/109541#issuecomment-1294241417
+    final TextStyle labelStyle = TextStyle(color: Color(color++));
+    final TextStyle unselectedLabelStyle = TextStyle(color: Color(color++));
+
+    final TabBarTheme materialTabBarTheme = TabBarTheme(
+      labelColor: buildMSC(),
+      unselectedLabelColor: Color(color++),
+      labelStyle: TextStyle(color: buildMSC()),
+      unselectedLabelStyle: TextStyle(color: Color(color++)),
+    );
+    final TabBarTheme tabBarTheme = TabBarTheme(
+      labelColor: Color(color++),
+      unselectedLabelColor: Color(color++),
+      labelStyle: TextStyle(color: Color(color++)),
+      unselectedLabelStyle: TextStyle(color: Color(color++)),
+    );
+    final TabBarTheme tabBarThemeWithNullUnselectedLabelColor = TabBarTheme(
+      labelColor: Color(color++),
+      labelStyle: TextStyle(color: Color(color++)),
+      unselectedLabelStyle: TextStyle(color: Color(color++)),
+    );
+
+    Future<void> buildTabBar({
```

Comment:
This is OK, bug typically the pumpWidget call is inline; `await test.pumpWidget(buildTabBar(...))`

Reply:
hmm that makes sense.